### PR TITLE
Hawtjni: Shared_pointer support added in hawtjni

### PR DESCRIPTION
**Pull request by ossdev07**
shared Pointer support was missing in the existing library code. So we have added this support and its working fine.

### CODE REVIEW

**Comment by ossdev07** [comment]
@chirino can you please have a look on these changes

File: hawtjni-generator/src/main/java/org/fusesource/hawtjni/generator/StructsGenerator.java [hunk 1]
```diff
@@ -455,8 +489,15 @@ void generateSetFields(JNIClass clazz) {
                 if( field.isPointer() ) {
                     output("(intptr_t)");
                 }
+		if( field.isSharedPointer() ) {
+                    output("(&");
+                }if(field.isSharedPointer() ) {
```
Comment (chirino) [inline hawtjni-generator/src/main/java/org/fusesource/hawtjni/generator/StructsGenerator.java]: 2 if statements with the same condition?

**Comment by ossdev07** [comment]
@chirino duplicate condition is removed thanks for review ...

**Comment by ossdev07** [comment]
@chirino please review and merge pull request so that changes of rocksdbjni can also be merged

**Comment by ossdev07** [comment]
@chirino can you have a look on these changes .. as its important for providing shared pointer functionality  for rocksdbjni

**Comment by ossdev07** [comment]
@chirino can you please have a look on these changes

**Comment by gnodet** [comment]
Can you add an example usage of `shared_ptr` to the [hawtjni-example](https://github.com/fusesource/hawtjni/tree/master/hawtjni-example)  ? I fear it's somewhat broken.

**Comment by gnodet** [comment]
The PR has been reverted, please open a new PR with a working example.

**Pull request merged**